cfn-resources/search-index/cmd/resource/resource.go

```diff
@@ -34,12 +34,10 @@ func setup() {
 	util.SetupLogger("search-index")
 }
 
-var CreateRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.PubKey, constants.PvtKey}
-var ReadRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.IndexID, constants.PubKey, constants.PvtKey}
-var UpdateRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.IndexID, constants.PubKey, constants.PvtKey}
-var DeleteRequiredFields = []string{constants.GroupID, constants.ClusterName, constants.IndexID, constants.PubKey, constants.PvtKey}
-var ListRequiredFields = []string{constants.ProjectID, constants.ClusterName, constants.Database, constants.CollectionName,
-	constants.PubKey, constants.PvtKey}
+var CreateRequiredFields = []string{constants.ProjectID, constants.ClusterName}
```

Comment:
General note for once we GA, changing the parameter names would be a breaking change.